#!/usr/bin/env python
PACKAGE = "move_by_ft_wrist"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("rate", int_t, 0, "node rate",    10, 0,   100)

gen.add("send_head_goals", bool_t, 0, "send goals to real topic", False)
gen.add("send_wrist_goals", bool_t, 0, "send goals to real topic", False)

gen.add("use_marker_orientation", bool_t, 0, "use marker orientation for wrist roll", False)

gen.add("min_x", double_t, 0, "workspace min_x",    0.2, -2.0,   2.0)
gen.add("max_x", double_t, 0, "workspace max_x",    0.6, -2.0,   2.0)
gen.add("min_y", double_t, 0, "workspace min_y",    -0.5, -2.0,   2.0)
gen.add("max_y", double_t, 0, "workspace max_y",    0.5, -2.0,   2.0)
gen.add("min_z", double_t, 0, "workspace min_z",    0.0, -2.0,   2.0)
gen.add("max_z", double_t, 0, "workspace max_z",    0.6, -2.0,   2.0)

gen.add('initial_pose_x', double_t, 0, "", 0.4, -2.0,   2.0)
gen.add('initial_pose_y', double_t, 0, "", -0.2, -2.0,   2.0)
gen.add('initial_pose_z', double_t, 0, "", 0.2, -2.0,   2.0)

gen.add('initial_pose_head_x', double_t, 0, "", 1.0, -2.0,   2.0)
gen.add('initial_pose_head_y', double_t, 0, "", 0.0, -2.0,   2.0)
gen.add('initial_pose_head_z', double_t, 0, "", 0.6, -2.0,   2.0)

gen.add('initial_pose_roll_degrees', double_t, 0, "", 0.0, -180.0, 180)
gen.add('initial_pose_pitch_degrees', double_t, 0, "", 60.0, -180.0, 180)
gen.add('initial_pose_yaw_degrees', double_t, 0, "", 0.0, -180.0, 180)



gen.add("global_frame_id", str_t, 0, "global_frame_id for goals", "/world")
gen.add("topic_name", str_t, 0, "topic", "/aruco_filtered")

gen.add("goal_timeout", double_t, 0, "timeout in seconds for going back to a home pose",    20, 1.0,   200)


exit(gen.generate(PACKAGE, PACKAGE, "ArucoFilter"))